### DISCUSSION TITLE
Hero info window, add information about spell point regeneration and spell learning level

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#4687] [Ideas]
I suggest adding the number of points regenerated by the hero each day (secondary skill Mysticism and artifacts are taken into account) at the end of the line "Spell Points:".

For example, display:
(Hero regenerates 1 spell point per day.)

"Spell Points: 22/40 +1"

For the hero's spell learning level, I suggest adding a "Spell Level:" line indicating the maximum spell learning level between the "Spell Points:" line and the "Move Points:" line.

For example, display:
(Hero can learn spells up to level 5.)

"Spell Points: 22/40 +1
Spell Level: 5
Move points: 1000/1000"

Ideally, the information displayed in the hero's info window can be selected from a settings file, for those who do not want this information. :)

I know that the current window is too small for all this. :)
But it is the window that needs to be adapted to display the desired information, not the other way around.
The window should also automatically adjust its size according to the selected information to be displayed.

In fheroes2:

![Hero info window 01](https://user-images.githubusercontent.com/43583994/143012207-44492648-8620-4c7f-a25e-56f7e90d05b5.png)

**Edit 23/11/2021.**

Related discussion: https://github.com/ihhub/fheroes2/discussions/3330.

**Comment by Branikolog** [comment]
Yes. this window is too small. Also, it's not obvious, what does this "+1" means here.
What if we just add new sentence describing mana recovery at hero info window?
![image](https://user-images.githubusercontent.com/55348946/143040669-a2ff5250-7358-4891-9380-da73fcb0a42b.png)

**Comment by zenseii** [comment]
> Yes. this window is too small. Also, it's not obvious, what does this "+1" means here. What if we just add new sentence describing mana recovery at hero info window? 


Wouldn't it just be as convienient to right click the mysticism skill to get to the same information then? I think the idea of LeHerosInconnu is to have this info readily available by just right clicking the hero portrait from the adventure map.

**Reply by a1exsh** [reply]
Additionally, there are artifacts that affect spell point regeneration, so just checking the skill doesn't give you the full picture.

**Reply by zenseii** [reply]
I've tried to figure out which artifacts do that in Heroes 2, but I can't find any artifacts that do.

**Reply by LeHerosInconnu** [reply]
The "Power Ring" artifact returns 2 extra spell points per turn to the hero.

**Comment by LeHerosInconnu** [comment]
Hello @Branikolog, @zenseii and @a1exsh,

> Yes. this window is too small. Also, it's not obvious, what does this "+1" means here.

This should be indicated in the manual, and after a few games it should be integrated by the player. :)

> What if we just add new sentence describing mana recovery at hero info window?
> ![image](https://user-images.githubusercontent.com/55348946/143040669-a2ff5250-7358-4891-9380-da73fcb0a42b.png)

The player just needs to right click on the hero on the adventure map to get the information.
No need to open the hero's screen, open a window to see the information, close the hero's screen to go back to the adventure map.
So only one clic.
And for this window, I have already planned additional information (as well as for other windows), just not posted yet. :)

> > Yes. this window is too small. Also, it's not obvious, what does this "+1" means here. What if we just add new sentence describing mana recovery at hero info window?
> 
> Wouldn't it just be as convienient to right click the mysticism skill to get to the same information then?

Both ways have to be possible.
And I have also planned additional information for Mysticism info window. :)

> I think the idea of LeHerosInconnu is to have this info readily available by just right clicking the hero portrait from the adventure map.

You get the idea. :)

> Additionally, there are artifacts that affect spell point regeneration, so just checking the skill doesn't give you the full picture.

This is planned for the Mysticism info window too. :)

> I've tried to figure out which artifacts do that in Heroes 2, but I can't find any artifacts that do.

The "Power Ring" artifact returns 2 extra spell points per turn to the hero.

**Reply by Branikolog** [reply]
Heroes are not always own "Mysticism" as a secondary skill. For sure, while clicking on the skill we can also show current recovery, but in my personal opinion, spell point recovery speed is such a simple value, 90% cases has a default value of 1. It depends on the only secondary skill and the only artifact in game (which is quite easy to consider). Mana recovery is highly predictable and changes not really often.

Anyway, "+1" looks not obvious for the window we're talking about. I'd rather prefer one more line with separate item: "Mana Recovery - 2 pts"
Or at least, regarding the proposal above, "+1/day", so player would understand more clearly, what does that +1 means.

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> Heroes are not always own "Mysticism" as a secondary skill. For sure, while clicking on the skill we can also show current recovery, but in my personal opinion, spell point recovery speed is such a simple value, 90% cases has a default value of 1. It depends on the only secondary skill and the only artifact in game (which is quite easy to consider). Mana recovery is highly predictable and changes not really often.

Precisely, the player can have several heroes, some with Mysticism, some without, some with an artifact that acts on Mysticism (or several, we should also think about the future of the game).
It is not so easy to remember all this.

> Anyway, "+1" looks not obvious for the window we're talking about. I'd rather prefer one more line with separate item: "Mana Recovery - 2 pts" Or at least, regarding the proposal above, "+1/day", so player would understand more clearly, what does that +1 means.

I also thought of displaying it like this:

"Spell Points: 22/40/+1"

But I find that the total spell points is not as discernible as before.

Or like this:

"Spell Points: +1/22/40"

But here, it's the number of remaining points that becomes less discernible.

Here: https://github.com/ihhub/fheroes2/discussions/3330, I had also thought of writing it on a separate line:

"Magic Restored: X"

When the option "Secondary Skills: Enhancement of some Secondary Skills" is enabled:

"Magic Restored: X-X" (or "Magic Restored: X/X")

The first number for the spell points restored per day.
The second number for spell points restored at the beginning of each combat.

But I think it should be on the same line as the spell points.

"+1/day" could be okay for the case above.
Although it would be a bit weird to have a term among numbers (on the right side of all colons in the window).
For me, "+1" would be explicit enough for the player after a few games.
However if "/day" is added, it will also be necessary to add "/combat" (with the mentioned option above activated).
And in this case a specific line should be added:

"Magic Restored: X/day - X/combat"

And then the line becomes very (too) long.

**Comment by LeHerosInconnu** [comment]
This is planned in my mod: https://github.com/ihhub/fheroes2/discussions/10100.
I close the discussion.
